src/auth/src/credentials/user.rs

```diff
@@ -13,10 +13,11 @@
 // limitations under the License.
 
 use crate::credentials::dynamic::CredentialsTrait;
-use crate::credentials::{Credentials, QUOTA_PROJECT_KEY, Result};
+use crate::credentials::{Credentials, Result};
 use crate::errors::{self, CredentialsError, is_retryable};
 use crate::token::{Token, TokenProvider};
 use crate::token_cache::TokenCache;
+use crate::utils::headers_util::build_headers;
```

Comment:
I would flatten these modules to `crate::header_utils::build_headers`.  As long as the module is not public this is enough, and we can always change it.

Reply:
This was actually my initial implementation before changing to the current one (for reasons similar to what I've described in my immediate previous comment). Will refactor accordingly. Curious to get thoughts from @harkamaljot and @sai-sunder-s.

Reply:
+1, we can flatten for now. We can always change it later if we decide to add more utils in future, since it is not public.